### DISCUSSION TITLE
Read config always at start, at init?

### DISCUSSION THREAD

**Discussion by kozaka-tv** [Th0rstenf/Rocksniffer-to-Streamer.bot-extension#100] [Ideas]
I do not know, how often this the case would be, but maybe at the beginning it will be a problem for the users.

If an user, has setup the app and created a timer, the first time the code runs, the Init will run once, and the configuration (GlobalVars) will be set and not red anymore. Like if I change some cooldown time from 3 to 5 for example, it will stay on 3 till I do not restart SB, or change the code and recompile and save.

How could we solve this problem? Read always the config? Have an Init action? Read regularry after each 10 seconds? Idk. Any ideas?

**Comment by Th0rstenf** [comment]
We already use the Init action from SB, which is executed only once at program start. 

I'd like to keep the instance alive, so we don't always have to instantiate everything from scratch. Even if it would probably not be noticable unless you look closely, just thinking about the unnecessary waste of processing time of instantiation and data base access for saving /restoring hurts my embedded guy brain :D 

But the actual reading of the config is really hardly noticable. I'll introduce an update config method that is called during Init() and cyclically, while leaving the instantiation of all objects and parsers in Init directly

**Comment by Th0rstenf** [comment]
In fact, I liked the idea so much that I already performed the change in #102
